### ISSUE TITLE
[GSD-13497] [XPU] DeltaNet reference kernel assertion failure in torch_chunk_gated_delta_rule

### ISSUE THREAD

**Issue by baustin27** [intel/compute-runtime#1002]
# Bug Report: Intel XPU DeltaNet reference kernel assertion failure in torch_chunk_gated_delta_rule

## Environment
- **Hardware**: Intel Arc Pro B70 Graphics (32GB VRAM)
- **OS**: Ubuntu 24.04 (Proxmox CT)
- **Compute Runtime**: 26.35.39758.10
- **PyTorch**: 2.14.0+xpu
- **Transformers**: 5.17.0 (includes `qwen3_5_moe` modeling)
- **Flash-linear-attention**: NOT installed (falls back to reference PyTorch implementation)

## Description
When running Qwen3.6-35B-A3B (a Gated DeltaNet hybrid architecture) on XPU, the model load succeeds but the first forward pass triggers an assertion failure in the reference implementation of `chunk_gated_delta_rule` (PyTorch fallback for `flash-linear-attention` which has no XPU kernels).

The assertion is in the SYCL tensor comparison kernel:
```
/__w/pytorch/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/TensorCompareKernels.cpp:193: operator(): global id: [0,0,0], local id: [0,0,0] Assertion `input_[0] != 0` failed.
```

This indicates a divide-by-zero in the reference kernel's triangular solve or l2norm path.

## Minimal Repro
```python
import torch
from transformers import AutoModelForCausalLM
from gptqmodel import GPTQModel, BACKEND, QuantizeConfig

# Use Qwen3.6 GPTQ-Int4 weights (any quantized Qwen3.6 works)
G = '/models/hub/models--palmfuture--Qwen3.6-35B-A3B-GPTQ-Int4/snapshots/00a66983516f8f8057741221277eed4141c9e431'
qc = QuantizeConfig.from_pretrained(G)
qc.device = 'xpu:0'
m = GPTQModel.from_pretrained(G, quantize_config=qc, backend=BACKEND.GPTQ_TORCH)

tok = m.tokenizer
p = tok.apply_chat_template([{'role':'user','content':'Reply with exactly one word: Paris'}], tokenize=False, add_generation_prompt=True)
ids = tok(p, return_tensors='pt').to('xpu:0')

with torch.no_grad():
    m.model(**ids)  # CRASHES HERE with SYCL assertion
```

## Stack Trace (abridged)
```
File ".../transformers/models/qwen3_5_moe/modeling_qwen3_5_moe.py", line 641
    core_attn_out, last_recurrent_state = torch_chunk_gated_delta_rule(
File ".../transformers/models/qwen3_5_moe/modeling_qwen3_5_moe.py", line 302
    @use_kernel_func_from_hub_with_fallback("chunk_gated_delta_rule", "fla")
    def torch_chunk_gated_delta_rule(...):
File ".../torch/native/xpu/sycl/TensorCompareKernels.cpp", line 193
    Assertion `input_[0] != 0` failed.
```

## Root Cause Analysis
The reference `torch_chunk_gated_delta_rule` in transformers performs:
1. `l2norm(query, dim=-1)` → `rsqrt(sum(x^2) + eps)`
2. Triangular solve: `torch.linalg.solve_triangular(ut_system, v_beta, upper=False, unitriangular=True)`
3. Cumulative decay: `cum_decay = decay.cumsum(dim=3)` then `cum_decay.exp()`

Any of these can produce zero denominators. The XPU SYCL implementation of `rsqrt`/`solve_triangular`/`exp` appears to hit an assert when input is zero, whereas CUDA silently produces inf/NaN.

## Workarounds Tried
1. **Install flash-linear-attention** → fails because it imports `torch.cuda` internally (CUDA-only)
2. **Force fp32 compute** → PyTorch XPU fp32 path works, but model is fp16 quantized; runtime conversion too slow
3. **Disable DeltaNet layers** → not possible without model surgery

## Impact
- Blocks all Gated DeltaNet hybrid models (Qwen3.6, Nemotron 3.5 Lightning, future Qwen3.8) on XPU
- The reference implementation is the *only* path since flash-linear-attention has no XPU backend
- Makes B70 unusable for the entire DeltaNet architecture class

## Request
1. **Fix the SYCL assertion** in `TensorCompareKernels.cpp` - zero input should produce inf/NaN per IEEE 754, not crash
2. **Provide XPU kernels for flash-linear-attention** (chunk_gated_delta_rule, fused_recurrent_gated_delta_rule)
3. **Document the fp32 requirement** for DeltaNet on XPU until kernels exist

## Additional Context
- The same model runs correctly on CPU (fp32 reference) and CUDA (flash-linear-attention optimized kernels)
- The bug is specific to the XPU reference fallback path
- Affected architectures: Qwen3.5 MoE, Qwen3.6, Nemotron 3.5 Lightning (Mamba-2 hybrid)

**Comment by kgibala** [comment]
Hi @baustin27,

Thank you for your contribution.

We kindly ask that issues be reported using the template described in the [Issue Submission Guide](https://github.com/intel/compute-runtime/blob/master/documentation/ISSUE_SUBMISSION_GUIDE.md). Without following this template, resolution time tends to be longer, as we need to ask follow-up questions to gather information that isn't included otherwise.

Could you please edit this issue to match the required format?

Thank you for your understanding.
